src/compiler/parser.ts

```diff
@@ -6069,7 +6069,7 @@ namespace ts {
                     const checkJsDirectiveMatchResult = checkJsDirectiveRegEx.exec(comment);
                     if (checkJsDirectiveMatchResult) {
                         checkJsDirective = {
-                            enabled: compareStrings(checkJsDirectiveMatchResult[1], "@ts-check", /*ignoreCase*/ true) === Comparison.EqualTo,
+                            enabled: equateStrings(checkJsDirectiveMatchResult[1], "@ts-check", /*ignoreCase*/ true),

```

Comment:
Can just call `equateStringsCaseInsensitive` directly.
But this case it might make more sense to just use ASCII capitalization (`.toLowerCase()`) instead of anything locale-sensitive.